.github/workflows/frontend.yaml

```diff
@@ -5,9 +5,70 @@ on:
   pull_request:
 
 jobs:
+  generate_schema:
+    uses: ./.github/workflows/generate-schema.yaml
+
+  compose_supergraph:
+    # Deduplicate jobs from pull requests and branch pushes within the same repo.
+    if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.repository
+    runs-on: ubuntu-latest
+    needs: generate_schema
+    steps:
```

Comment:
Can we use a reusable workflow here too? Looks like the commit stuff can all be placed after in the `federation` one